benchmark/webstreams/creation.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+const common = require('../common.js');
+const {
+  ReadableStream,
+  TransformStream,
+  WritableStream,
+} = require('node:stream/web');
+
+const bench = common.createBenchmark(main, {
+  n: [50e3],
+  kind: ['ReadableStream', 'TransformStream', 'WritableStream']
+});
+
+function main({ n, kind }) {
+  switch (kind) {
+    case 'ReadableStream':
+      new ReadableStream({});
```

Comment:
It's better to assign the value to a binding outside the loop and do something about it in the end, otherwise the optimizing compiler might be smart enough to perform a dead code elimination and make the benchmark pointless.

Reply:
A typical technique for micro-benchmarking JS would be to pre-fill an array before the measurement starts, and you assign the values into that array during the benchmark, then do something about the array in the end, in general the optimizing compiler won't attempt to eliminate the initializations in that case.

Reply:
Do we have a way to find dead code elimination? I tried some time ago with some V8 flags but it didn't work.

I've updated the code, does it solve? The benchmark results are the same.

Reply:
> Do we have a way to find dead code elimination? I tried some time ago with some V8 flags but it didn't work.

You can try `--trace-turbo*` flags with `--turbo-filter` to filter out specific functions you want to take a look at (and use v8/tools/turbolizer to help looking at the output). But I'd say to benchmark Node.js's JS code the best solution is to just avoid micro-benchmarks and test a comprehensive combination of operations with varied input ;)

Reply:
Indeed. I've created the creation.js to compare with `streams/creation.js`. This also shows how much memory it consumes (potential memory leak at some point IMHO - if you use the same `n` as `streams` it will get OOM).